.circleci/config.yml

```diff
@@ -9,6 +9,8 @@ jobs:
     <<: *defaults
     steps:
       - checkout
+      - setup_remote_docker:
```

Comment:
At first, we can split the test into unit tests and integration tests. Also, we need to specify multiple versions.

It's like
```
jobs:
  integration-test:
    parameters:
      docker_version:
        type: string
    steps:
      - checkout
      - setup_remote_docker
           version: << parameters.docker_version >>
      - run:
          name: Integration test
          command: make test-integration

integration_test_jobs: &integration_test_jobs
  - integration-test:
       name:  17.12.1-ce
       docker_version: 17.12.1-ce
  - integration-test:
       name:  18.01.0-ce
       docker_version: 18.01.0-ce
  - integration-test:
       name:  18.09.3
       docker_version: 18.09.3

workflows:
  version: 2
  integration-test:
    jobs: *integration_test_jobs
```

I don't think it works because this is just an example, but it's enough if you understand what I imagine.